### ISSUE TITLE
imgpkg pull time -b perf should not be influenced by number of referenced images

### ISSUE THREAD

**Issue by cppforlife** [carvel-dev/imgpkg#129]
**Describe the problem/challenge you have**

Currently `imgpkg pull -b` checks on every referenced image to rewrite images.yml. This may lead to slow down of imgpkg pull as bundle grows in number of referenced images. How can we decouple perf of imgpkg pull from number of referenced images.

(On a related topic, interesting use case: user is able to download bundle image but not authorized to download referenced images.)

**Comment by joaopapereira** [comment]
Sounds good to me. The stepping stone for this issue will be #136 due to this issue we will be able to with 1 request per bundle understand what needs to be pulled or not.

Accepting this issue.

**Comment by DennisDenuto** [comment]
```
bundle is pushed
bundle was copied (no locations img because bundle was copied by older imgpkg), images with same digest exist in the bundle repository
bundle was copied + locations img exists
```

case bundle was copied with older imgpkg
- if first refed img doesnt exist in bundle repository, then just use original img locations. no need to check original img locations for image existance
- Perhaps providing a warning message to the user "please copy the bundle again to make future pulls faster"

**Comment by DennisDenuto** [comment]
Blocked on https://github.com/vmware-tanzu/carvel-imgpkg/issues/161

EDIT: No longer blocked

**Comment by joaopapereira** [comment]
Fixed in version [0.12.0](https://github.com/vmware-tanzu/carvel-imgpkg/releases/tag/v0.12.0)
